packages/root-config/src/bootstrap/autocomplete/autocomplete.js

```diff
@@ -2,25 +2,27 @@ import {
   ServiceProvider,
   AutocompleteService,
   AutocompleteContextService,
-  RepositoryStorageService
+  RepositoryContextService
 } from '@ontotext/workbench-api';
 
 /**
  * Check if autocomplete is enabled, when loading(reloading) the application.
  * Gets the current autocomplete status from the backend and updates the context with the value.
  * If there is no selected repository, the request will not be made.
  */
-// TODO: Consider refactoring the code, adding a subscription to the selected repository change event, and updating the autocomplete status.
 const isAutocompleteEnabled = () => {
-  const currentRepository = ServiceProvider.get(RepositoryStorageService).getRepositoryReference();
-
-  if (!currentRepository.id) {
-    return Promise.resolve();
-  }
-  return ServiceProvider.get(AutocompleteService).isAutocompleteEnabled()
-    .then((enabled) => {
-      ServiceProvider.get(AutocompleteContextService).updateAutocompleteEnabled(enabled);
+  return new Promise((resolve, reject) => {
+    ServiceProvider.get(RepositoryContextService).onSelectedRepositoryChanged((selectedRepository) => {
+      if (!selectedRepository?.id) {
+        return resolve();
```

Comment:
I'm not sure, but should we set autocomplete to false when no repository is selected?